### PR TITLE
feat: Add support for the `user` object in Visibility Scripts

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

This adds basic support for the `user` object in visibility scripts. The goal was to have enough support for the scripts generated by the UI in the Admin Console. That would only require hasRole, but it wasn't too involved to add a few of the other basics quickly.

Also addressed a warning in the AdvancedSearchPage.test due to the togglePanel not correctly using `act()`.

I undertook manual testing with scripts testing various live combinations of roles and username etc.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by edalex-ian** [comment]
Ah rats, that'll teach me to sneak in an extra feature or two at the end. Fix on way.

**Comment by edalex-ian** [comment]
Ah, and now it has a conflict. :disappointed: Will go and solve that.

**Comment by edalex-ian** [comment]
Alright, should be good to go @PenghaiZhang :crossed_fingers:

File: react-front-end/tsrc/components/wizard/WizardHelper.tsx [hunk 1]
```diff
@@ -487,15 +488,30 @@ const buildXmlScriptObject = (values: PathValueMap): XmlScriptType =>
...
+  hasRole: (roleUniqueID) => userDetails.roles.includes(roleUniqueID),
+});
+
+/**
+ * Build a script context suitable for visibility scripts based on the current values of a wizard
+ * and a user's details.
+ * @param fieldValues the values of the various fields in the current wizard
+ * @param user details of typically the current user
+ */
+export const buildVisibilityScriptContext = (
```
Comment (PenghaiZhang) [inline react-front-end/tsrc/components/wizard/WizardHelper.tsx:508]: I wonder if these scripting related function should be placed in the `ScriptingModule`  ?

File: react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx [hunk 2]
```diff
@@ -75,13 +75,11 @@ mockGetAdvancedSearchByUuid.mockResolvedValue(getAdvancedSearchDefinition);
 
 const testUuid = "4be6ae54-68ca-4d8b-acd0-0ca96fc39280";
 
-const togglePanel = () =>
-  act(() => userEvent.click(screen.getByLabelText(filterButtonLabel)));
+const togglePanel = async () =>
+  act(async () => userEvent.click(screen.getByLabelText(filterButtonLabel)));
```
Comment (PenghaiZhang) [inline react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx:79]: Do we need a `await` ? I vaguely remember I asked this before but I forgot...

**Review by PenghaiZhang — APPROVED** [review]
Looks pretty nice and clear @edalex-ian

File: react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx (hunk 2 above)
Comment (edalex-ian) [reply react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx:79]: Nope, all good here. When I did it I checked with and without.

File: react-front-end/tsrc/components/wizard/WizardHelper.tsx (hunk 1 above)
Comment (edalex-ian) [reply react-front-end/tsrc/components/wizard/WizardHelper.tsx:508]: Wasn't sure, as it seemed this was a specific ScriptContext related to the way the Wizard handles it's values. I'm not sure if we may use it elsewhere or not... The line is a bit blurry here, it's definitely something I wondered about.

How about I leave it here for now and we'll see how it progresses?

Comment (PenghaiZhang) [reply react-front-end/tsrc/components/wizard/WizardHelper.tsx:508]: Good idea!

**Pull request merged**